pinot-core/src/test/java/org/apache/pinot/core/common/DataFetcherTest.java

```diff
@@ -363,7 +363,12 @@ public void testFetchBytesValues(String column) {
     _dataFetcher.fetchBytesValues(column, docIds, length, bytesValues);
 
     for (int i = 0; i < length; i++) {
-      Assert.assertEquals(new String(bytesValues[i], UTF_8), Integer.toString(_values[docIds[i]]), ERROR_MESSAGE);
+      String stringValue = Integer.toString(_values[docIds[i]]);
+      if (column != HEX_STRING_COLUMN && column != NO_DICT_HEX_STRING_COLUMN) {
```

Comment:
Instead of changing here, we should change `testFetchBytesValues()` to test on `STRING_COLUMN` and `NO_DICT_STRING_COLUMN` instead